File: config/environment.py
"""
Zentrale Python-Konfigurationsdatei für alle Services
Lädt die Umgebungskonfiguration basierend auf ENVIRONMENT
"""

import json
import os
from pathlib import Path

# Lade die Umgebungskonfiguration
config_path = Path(__file__).parent / 'environment.json'
with open(config_path, 'r', encoding='utf-8') as f:
    config = json.load(f)

# Bestimme die aktuelle Umgebung
environment = os.getenv('ENVIRONMENT', 'development')
current_config = config.get(environment, config['development'])

# Exportiere die Konfiguration
HOST = current_config['host']
PORTS = current_config['ports']
URLS = current_config['urls']
STORAGE = current_config['storage']

def get_service_url(service_name: str) -> str:
    """Hilfsfunktion um Service-URL zu erhalten"""
    return URLS.get(service_name, f"http://{HOST}:{PORTS[service_name]}")

def get_service_port(service_name: str) -> int:
    """Hilfsfunktion um Service-Port zu erhalten"""
    return PORTS[service_name]

def get_storage_path(storage_type: str) -> str:
    """Hilfsfunktion um Storage-Pfad zu erhalten"""
    return STORAGE.get(storage_type, STORAGE['basePath'])

File: config/environment.ts
/**
 * Zentrale TypeScript-Konfigurationsdatei für alle Services
 * Lädt die Umgebungskonfiguration basierend auf NODE_ENV
 */

import * as fs from 'fs';
import * as path from 'path';

// Lade die Umgebungskonfiguration
const configPath = path.join(__dirname, 'environment.json');
const envConfig = JSON.parse(fs.readFileSync(configPath, 'utf8'));

// Bestimme die aktuelle Umgebung
const environment = process.env.NODE_ENV || 'development';
const currentConfig = envConfig[environment] || envConfig.development;

// TypeScript-Interfaces
export interface ServiceConfig {
  host: string;
  ports: {
    backend: number;
    frontend: number;
    analyzer: number;
    audioService: number;
    audioSeparationService: number;
    saliencyService: number;
    visionService: number;
  };
  urls: {
    backend: string;
    frontend: string;
    analyzer: string;
    audioService: string;
    audioSeparationService: string;
    saliencyService: string;
    visionService: string;
  };
  storage: {
    basePath: string;
    audioStems: string;
    videos: string;
    keyframes: string;
  };
}

// Exportiere die Konfiguration
export const environmentConfig: ServiceConfig = {
  host: currentConfig.host,
  ports: currentConfig.ports,
  urls: currentConfig.urls,
  storage: currentConfig.storage,
} as ServiceConfig;

export const environmentName = environment;

// Hilfsfunktionen
export const getServiceUrl = (serviceName: keyof ServiceConfig['urls']): string => {
  return currentConfig.urls[serviceName] || `http://${currentConfig.host}:${currentConfig.ports[serviceName]}`;
};

export const getServicePort = (serviceName: keyof ServiceConfig['ports']): number => {
  return currentConfig.ports[serviceName];
};

export const getStoragePath = (storageType: keyof ServiceConfig['storage']): string => {
  return currentConfig.storage[storageType] || currentConfig.storage.basePath;
};
